private_storage/fields.py

```diff
@@ -64,7 +64,8 @@ def clean(self, *args, **kwargs):
     def generate_filename(self, instance, filename):
         subdir_func = self._upload
         if subdir_func:
-            subdirs = [self.storage.get_valid_name(dir) for dir in subdir_func(instance)]
```

Comment:
Some projects may actually expect to be able to return a `list` in the `subdir_func()`. Thus, having an `isinstance(dir, (list, tuple))` check here would be a great help!